### ISSUE TITLE
Configuration for training with full parameters

### ISSUE THREAD

**Issue by ICE-Bruce** [X-GenGroup/Flow-Factory#4]
Hello, I tried to train on qwen-image and zimage with 8 GPUs (80G each) using the full settings from the provided example config, but I got an OOM error. By the way, Lora setting will not get into OOM. Can you tell me how to run with 8 GPUs? Thank you.

**Comment by ICE-Bruce** [comment]
Hello, after adjusting the startup method, it can start successfully, but it can not converge well. Could you please provide the startup configuration file from the previous successful training?

**Comment by Jayce-Ping** [comment]
I previously tested Qwen-Image/Z-Image-Turbo on soly PickScore rewards and obtain the following training curves:

<img width="455" height="298" alt="Image" src="https://github.com/user-attachments/assets/6daa4b7a-a32b-4468-904c-6f8a1d659297" />

> Qwen-Image

<img width="452" height="302" alt="Image" src="https://github.com/user-attachments/assets/1de21c84-4fd6-4f9e-8df2-59c15f5b8646" />

> Z-Image-Turbo

Removing the `text_alignment` reward should be able to reproduce the results above. Would you like to show me your training curves so that I can know where the problem might be?

**Comment by ICE-Bruce** [comment]
<img width="948" height="478" alt="Image" src="https://github.com/user-attachments/assets/f4790124-1ab3-4a86-9c8a-93a213ebb2fb" />Hi， it's my training curve on z with my own reward model and I used the example file from this repo, like learning rate 1.0e-5, etc. This reward model has been validated as accurate through other methods. My only change was to modify the way the advantage was calculated, specifically by calculating the advantage in different dimensions separately and then weighting them.

`def compute_advantages(self, samples: List[BaseSample]) -> torch.Tensor:
    multi_rewards = torch.stack([s.extra_kwargs['multi_rewards'] for s in samples], dim=0)
    num_dims = multi_rewards.shape[1] 
    
    gathered_rewards = self.accelerator.gather(multi_rewards).cpu().numpy()

    unique_ids = torch.tensor([s.unique_id for s in samples], dtype=torch.int64, device=self.accelerator.device)
    gathered_ids = self.accelerator.gather(unique_ids).cpu().numpy()
    _unique_ids, group_indices = np.unique(gathered_ids, return_inverse=True)

    default_weights = [1.0] * num_dims
    weights = np.array(getattr(self.training_args, "reward_weights", default_weights), dtype=np.float64)
    
    total_advantages = np.zeros(len(gathered_rewards), dtype=np.float64)

    for group_id in np.unique(group_indices):
        group_mask_indices = (group_indices == group_id)
        group_res = gathered_rewards[group_mask_indices]
        
        group_weighted_adv = np.zeros(len(group_res), dtype=np.float64)

        for dim_idx in range(num_dims):
            dim_weight = weights[dim_idx]
            if dim_weight == 0:
                continue
            
            dim_scores = group_res[:, dim_idx]
            
            is_valid = (dim_scores != -1.0)
            valid_scores = dim_scores[is_valid]
            
            dim_adv = np.zeros_like(dim_scores)

            if len(valid_scores) > 1:
                val_mean = np.mean(valid_scores)
                val_std = np.std(valid_scores)
                
                if val_std < 1e-6:
                    pass 
                else:
                    normalized_scores = (valid_scores - val_mean) / (val_std + 1e-8)
                    dim_adv[is_valid] = normalized_scores
            
            group_weighted_adv += dim_adv * dim_weight

        total_advantages[group_mask_indices] = group_weighted_adv

    final_advantages = torch.as_tensor(total_advantages).reshape(
        self.accelerator.num_processes, -1
    )[self.accelerator.process_index].to(self.accelerator.device)

    for sample, adv in zip(samples, final_advantages):
        sample.extra_kwargs['advantage'] = adv

    return final_advantages`

**Comment by ICE-Bruce** [comment]
Not yet. Also, there's considerable variability in the model's inference results. For example, the average reward score within the group is 3.39, with a standard deviation of 0.36.

> That's is weird. Have you tried to switch to other models like Flux.1?

**Comment by Jayce-Ping** [comment]
I don't know if it is a problem of Z-Image-Turbo, since it has been RL tuned. I also notices drop when training LoRA with PickScore on this model. I suggest to switch to another model like Flux.1, which is 100% not RL tuned, to see if it is the problem of the model choice. Flux.1 has a 12B transformer and should be easy to train.
